libp2p/messagesHandler.go

```diff
@@ -562,13 +563,22 @@ func (handler *messagesHandler) ProcessReceivedMessage(message p2p.MessageP2P, f
 		handler.debugger.AddIncomingMessage(msg.Topic(), uint64(len(msg.Data())), !messageOk)
 
 		if messageOk {
-			handler.peersRatingHandler.IncreaseRating(fromConnectedPeer)
+			handler.increaseRatingIfNeeded(msg, fromConnectedPeer)
 		}
 	}(message)
 
 	return nil
 }
 
+func (handler *messagesHandler) increaseRatingIfNeeded(msg p2p.MessageP2P, fromConnectedPeer core.PeerID) {
```

Comment:
add a small unit test for all the conditions defined in this function?

Reply:
pushed